source/common/network/connection_impl.cc

```diff
@@ -810,6 +819,21 @@ void ServerConnectionImpl::onTransportSocketConnectTimeout() {
   closeConnectionImmediately();
 }
 
+ClientConnectionImpl::ClientConnectionImpl(
+    Event::Dispatcher& dispatcher, std::unique_ptr<Network::IoHandle> client_io_handle,
+    const Network::Address::InstanceConstSharedPtr& address,
+    const Network::Address::InstanceConstSharedPtr& source_address,
+    Network::TransportSocketPtr&& transport_socket,
+    const Network::ConnectionSocket::OptionsSharedPtr& options)
+    : ConnectionImpl(
+          dispatcher,
+          std::make_unique<ConnectionSocketImpl>(std::move(client_io_handle), nullptr, address),
+          std::move(transport_socket), stream_info_, false),
+      stream_info_(dispatcher.timeSource(), socket_->addressProviderSharedPtr()) {
+  UNREFERENCED_PARAMETER(source_address);
+  UNREFERENCED_PARAMETER(options);
```

Comment:
A drive-by comment: why these params are in the signature?

Reply:
Good catch!  This should be stripped from this PR. Update...
This is the early phase of internal connection PRs: I am not ready to provide the ability to create connection to internal listener.
The ultimate goal is to support whatever the below ClientConnectionImpl::ClientConnectionImpl could support